### PR TITLE
fix: Fix auto assignment after unassignment and ownership rule change

### PR DESCRIPTION
**Pull request by jangjodi**
Clear assignee_exists cache when an issue is unassigned so that issue owner can be recomputed
 
Fixes [ISSUE-1645](https://getsentry.atlassian.net/browse/ISSUE-1645)

[ISSUE-1645]: https://getsentry.atlassian.net/browse/ISSUE-1645?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/sentry/models/groupassignee.py [hunk 1]
```diff
@@ -111,6 +112,8 @@ def deassign(self, group: Group, acting_user: User | RpcUser | None = None) -> N
             Activity.objects.create_group_activity(group, ActivityType.UNASSIGNED, user=acting_user)
             record_group_history(group, GroupHistoryStatus.UNASSIGNED, actor=acting_user)
 
+            GroupOwner.invalidate_assignee_exists_cache(group.project.id)
```
Comment (jangjodi) [inline src/sentry/models/groupassignee.py:115]: Invalidate assignee exists cache when the issue is unassigned

File: src/sentry/tasks/post_process.py [hunk 2]
```diff
@@ -186,13 +189,17 @@ def handle_owner_assignment(job):
...
                         cache.set(
-                            assignee_key, assignees_exists, 60 * 60 * 24 if assignees_exists else 60
+                            assignee_key,
+                            assignees_exists,
+                            ASSIGNEE_EXISTS_DURATION
+                            if assignees_exists
+                            else ASSIGNEE_DOES_NOT_EXIST_DURATION,
                         )
 
                     if assignees_exists:
```
Comment (jangjodi) [inline src/sentry/tasks/post_process.py:205]: When the assignee exists cache is cleared, this `if` statement now does not pass and the early return does not occur

File: src/sentry/models/groupowner.py [hunk 3]
```diff
@@ -162,6 +165,24 @@ def invalidate_debounce_issue_owners_evaluation_cache(cls, project_id):
             cache_keys = [ISSUE_OWNERS_DEBOUNCE_KEY(group_id) for group_id in group_ids]
             cache.delete_many(cache_keys)
 
+    @classmethod
+    def invalidate_assignee_exists_cache(cls, project_id):
+        # Get all the groups for a project that had an event within the ASSIGNEE_EXISTS_DURATION window.
```
Comment (snigdhas) [inline src/sentry/models/groupowner.py:170]: Great, succinct comment!

File: tests/sentry/tasks/test_post_process.py [hunk 4]
```diff
@@ -936,6 +941,71 @@ def test_owner_assignment_when_owners_have_been_unassigned(self):
...
+        )
+
+        self.call_post_process_group(
+            is_new=False,
+            is_regression=False,
+            is_new_group_environment=False,
+            event=event,
+        )
+
+        assignee = assignee = (
```
Comment (snigdhas) [inline tests/sentry/tasks/test_post_process.py:967]: Typo?

**Review by snigdhas — APPROVED** [review]

File: tests/sentry/tasks/test_post_process.py (hunk 4 above)
Comment (jangjodi) [reply tests/sentry/tasks/test_post_process.py]: thank you!

**Pull request merged**